### PR TITLE
Adds more logging to test_unique_host_constraint

### PR DESCRIPTION
**Pull request by dposada**
## Changes proposed in this PR

- adding a log with the number of total, running, and waiting jobs

## Why are we making these changes?

To make failures easier to track down.

### CODE REVIEW

File: integration/tests/cook/test_basic.py [hunk 1]
```diff
@@ -1640,14 +1640,16 @@ def num_running_predicate(response):
...
+                                 f'and {num_waiting} waiting job(s)')
                 if num_jobs_total == num_hosts + 1:
                     # One job should not be scheduled
                     return (num_running == num_jobs_total - 1) and (num_waiting == 1)
                 else:
                     # All of the jobs should be running
                     return num_running == num_jobs_total
 
-            jobs = util.wait_until(query, num_running_predicate, max_wait_ms=60000)
+            jobs = util.wait_until(query, num_running_predicate)
```
Comment (pschorf) [inline integration/tests/cook/test_basic.py]: Why did the timeout change?

Comment (dposada) [reply integration/tests/cook/test_basic.py]: 60 seconds isn't enough in some of our environments, and if you use the default, you can change it with an environment variable, which is nice

Comment (pschorf) [reply integration/tests/cook/test_basic.py]: 👍

**Comment by pschorf** [comment]
Build is red

**Pull request merged**